### DISCUSSION TITLE
Incoming http/s request to Actions from outside world?

### DISCUSSION THREAD

**Discussion by sgoff0** [community/community#26077] [Actions]
<p>Within GitHub Actions can service containers (or anything else in a workflow) receive network requests from the outside world?</p>
<p>Specifically I’m using Appium (client/server architecture) for mobile testing.  Ideally within GitHub Actions I’d spin up a service container with a mock server and then start my mobile tests, sending commands to a third party responsible for executing on a device.  Since this device is running on a different network than GitHub Action’s workflow, is there any way to make requests from the third party into my workflow’s service container?  If not using service containers is there any other way to achieve this within Actions?</p>
<p>Per documentation I believe incoming network requests are only possible if everything is local to GitHub Actions but I’d like to confirm.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/sgoff0">@sgoff0</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>What come to my mind is exposing the runner to public with <a href="https://ngrok.com/" rel="nofollow noopener">ngrok</a>.</p>
<p>I’m not familar with Appium, i only tried to post webhook payload to internal server(ngrok configured) before.</p>
<p>Could you please follow the <a href="https://dashboard.ngrok.com/get-started/setup" rel="nofollow noopener">setupdoc</a> and use the url (eg: http://***.ngrok.io) to receive the request for a try?</p>
<p>Thanks</p>
